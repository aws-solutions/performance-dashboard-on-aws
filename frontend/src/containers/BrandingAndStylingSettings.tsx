/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { useHistory } from "react-router-dom";
import { useTranslation } from "react-i18next";
import SettingsLayout from "../layouts/Settings";
import Button from "../components/Button";
import "./PublishingGuidanceSettings.css";
import Logo from "../components/Logo";
import Favicon from "../components/Favicon";
import { useSettings } from "../hooks";
import Spinner from "../components/Spinner";

function BrandingAndStylingSettings() {
    const history = useHistory();
    const { settings } = useSettings();
    const { t } = useTranslation();

    const onEditLogo = () => {
        history.push("/admin/settings/brandingandstyling/editlogo");
    };

    const onEditFavicon = () => {
        history.push("/admin/settings/brandingandstyling/editfavicon");
    };

    const onEditColors = () => {
        history.push("/admin/settings/brandingandstyling/editcolors");
    };

    return (
        <SettingsLayout>
            <h1>{t("BrandingAndStyling")}</h1>

            <p>{t("BrandingAndStylingDescription")}</p>
            <br />

            <h2 className="margin-top-2-important">{t("BrandingAndStylingLogo")}</h2>
            <p>{t("BrandingAndStylingLogoDescription")}</p>
            <br />

            <div className="grid-row margin-top-0-important">
                <div className="grid-col flex-1 text-left">
                    <Logo />
                </div>
                <div className="grid-col flex-9 padding-left-3 text-right">
                    <Button className="margin-top-2" variant="outline" onClick={onEditLogo}>
                        {t("Edit")}
                    </Button>
                </div>
            </div>

            <hr
                style={{
                    border: "none",
                    height: "1px",
                    backgroundColor: "#dfe1e2",
                    margin: "2rem 0",
                }}
            />

            <h2 className="margin-top-2-important">{t("BrandingAndStylingFavicon")}</h2>
            <p>{t("BrandingAndStylingFaviconDescription")}</p>
            <br />

            <div className="grid-row margin-top-0-important">
                <div className="grid-col flex-1 text-left">
                    <Favicon />
                </div>
                <div className="grid-col flex-9 padding-left-3 text-right">
                    <Button className="margin-top-2" variant="outline" onClick={onEditFavicon}>
                        {t("Edit")}
                    </Button>
                </div>
            </div>

            <hr
                style={{
                    border: "none",
                    height: "1px",
                    backgroundColor: "#dfe1e2",
                    margin: "2rem 0",
                }}
            />

            <h2 className="margin-top-2-important">{t("BrandingAndStylingColors")}</h2>
            <p>{t("BrandingAndStylingColorsDescription")}</p>
            <br />

            <div className="grid-row margin-top-0-important">
                <div className="grid-col flex-1 text-left">
                    <div className="grid-col flex-1 text-left">
                        <h3>{t("BrandingAndStylingPrimaryColor")}</h3>
                    </div>
                    {settings && settings.colors ? (
                        <div className="grid-row margin-top-0-important">
                            <div
                                className="radius-md"
                                style={{
                                    backgroundColor: settings.colors.primary,
                                    margin: "10px 0px",
                                    width: 25,
                                    height: 25,
                                }}
                            ></div>
                            <div className="font-sans-lg grid-col flex-9 margin-left-1 margin-top-05">
                                {settings.colors.primary}
                            </div>
                        </div>
                    ) : (
                        <Spinner
                            className="margin-top-3 text-center"
                            label={t("LoadingSpinnerLabel")}
                        />
                    )}
                </div>
                <div className="grid-col flex-3 grid-col flex-3 text-right">
                    <Button className="margin-top-2" variant="outline" onClick={onEditColors}>
                        {t("Edit")}
                    </Button>
                </div>
            </div>
        </SettingsLayout>
    );
}

export default BrandingAndStylingSettings;
