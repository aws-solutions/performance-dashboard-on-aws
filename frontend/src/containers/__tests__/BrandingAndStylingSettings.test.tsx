/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import BrandingAndStylingSettings from "../BrandingAndStylingSettings";

jest.mock("../../hooks");

test("renders the title", async () => {
    const { getByRole } = render(<BrandingAndStylingSettings />, {
        wrapper: MemoryRouter,
    });
    expect(getByRole("heading", { name: "Branding and styling" })).toBeInTheDocument();
});

test("renders the description", async () => {
    const { getByText } = render(<BrandingAndStylingSettings />, {
        wrapper: MemoryRouter,
    });
    expect(getByText("Customize your performance dashboard.")).toBeInTheDocument();
});

test("renders the logo title", async () => {
    const { getByRole } = render(<BrandingAndStylingSettings />, {
        wrapper: MemoryRouter,
    });
    expect(getByRole("heading", { name: "Logo" })).toBeInTheDocument();
});

test("renders the logo description", async () => {
    const { getByText } = render(<BrandingAndStylingSettings />, {
        wrapper: MemoryRouter,
    });
    expect(
        getByText(
            "This logo will appear in the header next to the performance dashboard name and in the published site header.",
        ),
    ).toBeInTheDocument();
});

test("renders the colors title", async () => {
    const { getByRole } = render(<BrandingAndStylingSettings />, {
        wrapper: MemoryRouter,
    });
    expect(getByRole("heading", { name: "Colors" })).toBeInTheDocument();
});

test("renders the colors description", async () => {
    const { getByText } = render(<BrandingAndStylingSettings />, {
        wrapper: MemoryRouter,
    });
    expect(
        getByText(
            "Customize these colors to make your dashboards appear similar in style to your organization's brand and color palette.",
        ),
    ).toBeInTheDocument();
});

test("renders two  buttons to edit", async () => {
    const { getAllByRole } = render(<BrandingAndStylingSettings />, {
        wrapper: MemoryRouter,
    });
    const buttons = getAllByRole("button", { name: "Edit" });
    expect(buttons[0]).toBeInTheDocument();
    expect(buttons[1]).toBeInTheDocument();
});

test("branding and styling settings should match snapshot", async () => {
    const wrapper = render(<BrandingAndStylingSettings />, {
        wrapper: MemoryRouter,
    });
    expect(wrapper.container).toMatchSnapshot();
});
