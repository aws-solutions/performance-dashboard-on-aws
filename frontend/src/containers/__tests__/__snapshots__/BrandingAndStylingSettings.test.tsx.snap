// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`branding and styling settings should match snapshot 1`] = `
<div>
  <nav
    aria-label="Breadcrumbs"
    class="usa-breadcrumb padding-top-0"
  >
    <ol
      class="usa-breadcrumb__list"
    >
      <li
        class="usa-breadcrumb__list-item"
      >
        <a
          aria-label=""
          class="link"
          href="/admin/settings/topicarea"
          target=""
        >
          <span>
            <span>
              Settings
            </span>
          </span>
        </a>
      </li>
      <li
        aria-current="page"
        class="usa-breadcrumb__list-item usa-current cursor-default"
      >
        <span>
          Ministries
        </span>
      </li>
    </ol>
  </nav>
  <div
    class="usa-section padding-top-3"
  >
    <div
      class="grid-container padding-x-0"
    >
      <div
        class="grid-row grid-gap"
      >
        <div
          class="desktop:grid-col-3"
        >
          <nav
            aria-label="Left navigation"
          >
            <ul
              class="usa-sidenav"
            >
              <li
                class="usa-sidenav__item"
              >
                <a
                  class="usa-current"
                  href="/admin/settings"
                >
                  Settings
                </a>
                <ul
                  class="usa-sidenav__sublist"
                >
                  <li
                    class="usa-sidenav__item"
                  >
                    <a
                      class="usa-current"
                      href="/admin/settings/topicarea"
                    >
                      Ministries
                    </a>
                  </li>
                </ul>
                <ul
                  class="usa-sidenav__sublist"
                >
                  <li
                    class="usa-sidenav__item"
                  >
                    <a
                      class=""
                      href="/admin/settings/publishingguidance"
                    >
                      Publishing guidance
                    </a>
                  </li>
                </ul>
                <ul
                  class="usa-sidenav__sublist"
                >
                  <li
                    class="usa-sidenav__item"
                  >
                    <a
                      class=""
                      href="/admin/settings/brandingandstyling"
                    >
                      Branding and styling
                    </a>
                  </li>
                </ul>
                <ul
                  class="usa-sidenav__sublist"
                >
                  <li
                    class="usa-sidenav__item"
                  >
                    <a
                      class=""
                      href="/admin/settings/publishedsite"
                    >
                      Published site
                    </a>
                  </li>
                </ul>
                <ul
                  class="usa-sidenav__sublist"
                >
                  <li
                    class="usa-sidenav__item"
                  >
                    <a
                      class=""
                      href="/admin/settings/dateformat"
                    >
                      Date and time format
                    </a>
                  </li>
                </ul>
                <ul
                  class="usa-sidenav__sublist"
                >
                  <li
                    class="usa-sidenav__item"
                  >
                    <a
                      class=""
                      href="/admin/settings/adminsite"
                    >
                      Admin site
                    </a>
                  </li>
                </ul>
              </li>
            </ul>
          </nav>
        </div>
        <main
          class="desktop:grid-col-9 usa-prose"
          id="main-content"
        >
          <h1>
            Branding and styling
          </h1>
          <p>
            Customize your performance dashboard.
          </p>
          <br />
          <h3
            class="margin-top-2-important"
          >
            Logo
          </h3>
          <p>
            This logo will appear in the header next to the performance dashboard name and in the published site header.
          </p>
          <br />
          <div
            class="grid-row margin-top-0-important"
          >
            <div
              class="grid-col flex-1 text-left"
            >
              <img
                alt="Organization logo"
                hidden=""
                src="logo.svg"
              />
            </div>
            <div
              class="grid-col flex-9 padding-left-3"
            >
              <button
                class="usa-button usa-button--outline margin-top-2"
              >
                Edit
              </button>
            </div>
          </div>
          <hr
            style="height: 1px; background-color: rgb(223, 225, 226); margin: 2rem 0px;"
          />
          <h3
            class="margin-top-2-important"
          >
            Colors
          </h3>
          <p>
            Customize these colors to make your dashboards appear similar in style to your organization's brand and color palette.
          </p>
          <br />
          <div
            class="grid-row margin-top-0-important"
          >
            <div
              class="grid-col flex-1 text-left"
            >
              <div
                class="text-bold grid-col flex-1 text-left"
              >
                Primary color
              </div>
              <div
                class="grid-row margin-top-0-important"
              >
                <div
                  class="radius-md"
                  style="background-color: rgb(255, 255, 255); margin: 10px 0px; width: 25px; height: 25px;"
                />
                <div
                  class="font-sans-lg grid-col flex-9 margin-left-1 margin-top-05"
                >
                  #ffffff
                </div>
              </div>
            </div>
            <div
              class="grid-col flex-3 grid-col flex-3 text-right"
            >
              <button
                class="usa-button usa-button--outline margin-top-2"
              >
                Edit
              </button>
            </div>
          </div>
        </main>
      </div>
    </div>
  </div>
</div>
`;
